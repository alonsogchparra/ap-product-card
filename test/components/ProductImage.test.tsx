import React from 'react';
import { render } from '@testing-library/react';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../../src/data/products';

describe('ProductImage', () => {
  test('should render component correctly with custom image', () => {
    const { asFragment } = render(<ProductImage img="./custom-image.png" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should show the component with the product image', () => { 
    const {asFragment} = render(<ProductCard product={product2}>{() => <ProductImage />}</ProductCard>);

    expect(asFragment()).toMatchSnapshot();
   })
});
