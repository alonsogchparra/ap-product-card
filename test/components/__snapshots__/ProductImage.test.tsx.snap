// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should render component correctly with custom image 1`] = `
<DocumentFragment>
  <img
    alt="Product Image"
    class="productImg undefined"
    src="./custom-image.png"
  />
</DocumentFragment>
`;

exports[`ProductImage should show the component with the product image 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <img
      alt="Product Image"
      class="productImg undefined"
      src="./coffee-mug2.png"
    />
  </div>
</DocumentFragment>
`;
